python/src/aiconfig/editor/server/server.py

```diff
@@ -211,6 +247,45 @@ async def run() -> FlaskResponse:
         ).to_flask_format()
 
 
+@app.route("/cancel", methods=["POST"])
+def cancel() -> FlaskResponse:
+    state = get_server_state(app)
+    request_json = request.get_json()
+
+    cancellation_token_id: Union[str, None] = request_json.get("cancellation_token_id")
+    if cancellation_token_id is not None:
+        task = state.tasks.get(cancellation_token_id)
+        if task is not None:
+            # TODO: saqadri - specify a custom message here if needed
+            task.cancel()
```

Comment:
I don't think this is going to work unless the task being canceled awaits frequently. In order to actually yield control back, it has to await something.

Reply:
You're right :). Updated implementation